packages/transaction-controller/src/utils.ts

```diff
@@ -36,10 +37,9 @@ const NORMALIZERS: { [param in keyof TransactionParams]: any } = {
  */
 export function normalizeTxParams(txParams: TransactionParams) {
   const normalizedTxParams: TransactionParams = { from: '' };
-  let key: keyof TransactionParams;
-  for (key in NORMALIZERS) {
-    if (txParams[key as keyof TransactionParams]) {
-      normalizedTxParams[key] = NORMALIZERS[key](txParams[key]) as never;
+  for (const key of getKnownPropertyNames(NORMALIZERS)) {
+    if (txParams[key]) {
+      normalizedTxParams[key] = NORMALIZERS[key](txParams[key]);
     }
   }
   return normalizedTxParams;
```

Comment:
When inspecting property keys with signatures that are more specific than `string`, use `getKnownPropertyNames()` from `@metamask/utils` as a safe alternative to `for..in`.